molgenis-api-metadata/src/main/java/org/molgenis/api/metadata/v3/MetadataApiServiceImpl.java

```diff
@@ -117,29 +118,33 @@ public Attributes findAttributes(
   @Override
   public Attribute findAttribute(String entityTypeId, String attributeId) {
     EntityType entityType = findEntityType(entityTypeId);
+    return findAttribute(entityType, attributeId);
+  }
 
+  private Attribute findAttribute(EntityType entityType, String attributeId) {
     // TODO use MetaDataService instead of DataService
     Attribute attribute =
         dataService.findOneById(
             AttributeMetadata.ATTRIBUTE_META_DATA, attributeId, Attribute.class);
-    if (attribute == null) {
+    if (attribute == null || !attribute.getEntity().getId().equals(entityType.getId())) {
```

Comment:
Can't we just use ```entityType.getOwnAttribute(attributeId)```?

Reply:
You can only use `entityType.getAttribute` to get a single attribute but that also returns parent attributes.